src/keys.h

```diff
@@ -43,58 +43,118 @@ enum {
     EV_CODE_MOUSE_OFF3
 };
 
-/* The low 16 bits of the MODS is the standard X modifier mask. Bits
-   16 to 20 define the type of the event. One of the following: */
-
 enum {
-    EV_TYPE_KEY   = 0x00010000,
-    EV_TYPE_MOUSE = 0x00020000,
+    /*
+      The low 16 bits of the MODS are the standard X modifier mask, as
+      found in <X11/X.h>.  The highest standard modifier mask (1<<15)
+      corresponds to AnyModifier, but there's a gap for (1<<13) and
+      (1<<14) that we can fill.  According to libinput(4):
+
+        X clients receive events with logical button numbers, where 1, 2, 3 are
+        usually interpreted as left, middle, right and logical buttons 4, 5, 6,
+        7 are usually interpreted as scroll up, down, left, right.  The  fourth
+        and fifth physical buttons on a device will thus send logical buttons 8
+        and 9.
+
+      These button 6/7 masks are probably consistent with everybody
+      everywhere.  They are definitely consistent with Tcl/Tk:
+      https://core.tcl-lang.org/tk/artifact/fe30190927f08326.
+    */
+    EV_BUTTON6_MASK = 0x00002000, /* the (1<<13) gap */
+    EV_BUTTON7_MASK = 0x00004000, /* the (1<<14) gap */
+
+    /*
+      Bits 16 to 20 define the type of the event.  We allocate two right
+      now.  We have two available.
```

Comment:
Grr.  This should be as follows:

> Bits 16 and 17 define event types.  EV_TYPE_MASK has to cover all event types.

I can fix that in a follow-up.

Reply:
Why don't you fix it now and simply amend it? Would keep the history clean.

Reply:
I know how to make the change to my desktop git repo.  (Create the commit, rebase interactively to move it ahead of the not-yet-pull-requested unrelated fix and combine.)

That will leave Github with a couple of completely different heads and commits.  If it's possible to fix that broken situation, I don't know how.